src/parser/document-builder.ts

```diff
@@ -87,9 +84,11 @@ export function createDocument(filePath: string, tokens: Node[], reporting: Repo
 					if (bindings.length) {
 						for(const b of bindings)
 							allBindings.add(b.bindingHandler.name)
-						const currentAstNode = astNodeStack[astNodeStack.length - 1]
-						const bindingNode = new BindingNode(currentAstNode, bindings)
-						currentAstNode.childNodes.push(bindingNode)
+						const currentContext = astContextStack[astContextStack.length - 1]
+						if (!currentContext)
+							throw new Diagnostic(filePath, 'no-viewmodel-reference', Object.assign({}, node.loc))
```

Comment:
This was the big change to get it to work. Had missed guarding for this case for the self-closing tags.